nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/xml/XMLRecordSetWriter.java

```diff
@@ -0,0 +1,196 @@
+import java.util.List;
+
+@Tags({"xml", "resultset", "writer", "serialize", "record", "recordset", "row"})
+@CapabilityDescription("Writes a RecordSet to XML. The records are wrapped by a root tag.")
+public class XMLRecordSetWriter extends DateTimeTextRecordSetWriter implements RecordSetWriterFactory {
+
+    public static final AllowableValue ALWAYS_SUPPRESS = new AllowableValue("always-suppress", "Always Suppress",
+            "Fields that are missing (present in the schema but not in the record), or that have a value of null, will not be written out");
+    public static final AllowableValue NEVER_SUPPRESS = new AllowableValue("never-suppress", "Never Suppress",
+            "Fields that are missing (present in the schema but not in the record), or that have a value of null, will be written out as a null value");
+    public static final AllowableValue SUPPRESS_MISSING = new AllowableValue("suppress-missing", "Suppress Missing Values",
+            "When a field has a value of null, it will be written out. However, if a field is defined in the schema and not present in the record, the field will not be written out.");
+
+    public static final AllowableValue USE_PROPERTY_AS_WRAPPER = new AllowableValue("use-property-as-wrapper", "Use Property as Wrapper",
+            "The value of the property \"Array Tag Name\" will be used as the tag name to wrap elements of an array. The field name of the array field will be used for the tag name " +
+                    "of the elements.");
+    public static final AllowableValue USE_PROPERTY_FOR_ELEMENTS = new AllowableValue("use-property-for-elements", "Use Property for Elements",
+            "The value of the property \"Array Tag Name\" will be used for the tag name of the elements of an array. The field name of the array field will be used as the tag name " +
+                    "to wrap elements.");
+    public static final AllowableValue NO_WRAPPING = new AllowableValue("no-wrapping", "No Wrapping",
+            "The elements of an array will not be wrapped");
+
+    public static final PropertyDescriptor SUPPRESS_NULLS = new PropertyDescriptor.Builder()
+            .name("suppress_nulls")
+            .displayName("Suppress Null Values")
+            .description("Specifies how the writer should handle a null field")
+            .allowableValues(NEVER_SUPPRESS, ALWAYS_SUPPRESS, SUPPRESS_MISSING)
+            .defaultValue(NEVER_SUPPRESS.getValue())
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor PRETTY_PRINT_XML = new PropertyDescriptor.Builder()
+            .name("pretty_print_xml")
+            .displayName("Pretty Print XML")
+            .description("Specifies whether or not the XML should be pretty printed")
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .required(true)
+            .build();
+

```

Comment:
EL currently is not considered for RecordSetWriters, right?

Reply:
If you want to add it, @mattyb149 did an API update for the DBCPService that [shows how](https://github.com/apache/nifi/commit/099bfcdf3a5873a311312eb7e9e85b7b22ef1b98). I would consider it optional, though, because AFAIK the issue hasn't been given much discussion.